bot/admin/server/src/main/kotlin/BotAdminService.kt

```diff
@@ -153,6 +153,48 @@ object BotAdminService {
         }
     }
 
+    fun saveAnnotation(
+        dialogId: String,
+        actionId: String,
+        annotationDTO: BotAnnotationDTO,
+        user: String
+    ): BotAnnotation {
+        return if (dialogReportDAO.annotationExists(dialogId, actionId)) {
+            updateAnnotation(dialogId, actionId, annotationDTO, user)
+        } else {
+            createAnnotationInternal(dialogId, actionId, annotationDTO, user)
+        }
+    }
+
+    private fun createAnnotationInternal(
```

Comment:
Why internal? Maybe I'm wrong, but I think you can simply call `createAnnotation` and not create a second function for the same thing. Maybe I'm wrong.

Reply:
in all cases, you need to make the suggested code improvement, whether for createAnnotationInternal or createAnnotation